internal/cli/commands/book.go

```diff
@@ -59,64 +67,91 @@ func run(cmd *cobra.Command, args []string, getBS func() booking.BookingService,
 		available = rankings.Sort(available)
 	}
 
-	var filters []filter.RoomFilter
-	if cmd.Flags().Changed(CampusFlagName) {
-		filters = append(filters, getCampusFilter(campus))
-	}
-	if cmd.Flags().Changed(SizeFlagName) {
-		filters = append(filters, getSizeFilter(roomSize))
-	}
+	var n int
 
-	available = filter.Filter(available, filters)
+	if !cmd.Flags().Changed(RoomFlagName) {
+		var filters []filter.RoomFilter
+		if cmd.Flags().Changed(CampusFlagName) {
+			filters = append(filters, getCampusFilter(campus))
+		}
+		if cmd.Flags().Changed(SizeFlagName) {
+			filters = append(filters, getSizeFilter(roomSize))
+		}
 
-	showAvailable(available, cmd.Flags().Changed(SizeFlagName))
+		available = filter.Filter(available, filters)
 
-	room, err := prompt("Room to book")
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
-	}
+		showAvailable(available, cmd.Flags().Changed(SizeFlagName))
 
-	n, err := strconv.Atoi(room)
-	n--
-	if err != nil {
-		fmt.Printf("Invalid Room\n")
-		os.Exit(1)
-	}
+		room, err := prompt("Room to book")
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
 
-	message, err := prompt("Message to add with the booking (default: empty)")
-	if err != nil {
-		fmt.Println(err)
-		fmt.Println("No booking was made")
-		os.Exit(1)
+		n, err = strconv.Atoi(room)
+		n--
+		if err != nil {
+			fmt.Printf("Invalid Room\n")
+			os.Exit(1)
+		}
 	}
 
-	if n < len(available) && n >= 0 {
-		fmt.Printf("Booking %s...\n", available[n].Id)
-		booking := booking.Booking{
-			Room:  available[n],
-			Start: startDate,
-			End:   endDate,
-			Text:  message,
-		}
-		err := bs.Book(booking)
+	if !cmd.Flags().Changed(MessageFlagName) {
```

Comment:
Would maybe be more nice if the room flag was validated before a potential message promt